### PR TITLE
[browser] fix bit width from 32 to 8 for managed boolean

### PR DESCRIPTION
**Pull request by pavelsavara**

### CODE REVIEW

File: src/mono/browser/runtime/marshal.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ export function is_args_exception(args: JSMarshalerArguments): boolean {
 export function is_receiver_should_free(args: JSMarshalerArguments): boolean {
     if (!WasmEnableThreads) return false;
     mono_assert(args, "Null args");
-    return getB32(<any>args + JSMarshalerArgumentOffsets.ReceiverShouldFree);
+    return getB8(<any>args + JSMarshalerArgumentOffsets.ReceiverShouldFree);
```
Comment (pavelsavara) [inline src/mono/browser/runtime/marshal.ts:92]: I hope this is what causes https://github.com/dotnet/runtime/issues/96546

File: src/mono/browser/runtime/marshal.ts [hunk 2]
```diff
@@ -106,7 +106,7 @@ export function get_caller_native_tid(args: JSMarshalerArguments): PThreadPtr {
 
 export function set_receiver_should_free(args: JSMarshalerArguments): void {
     mono_assert(args, "Null args");
-    setB32(<any>args + JSMarshalerArgumentOffsets.ReceiverShouldFree, true);
+    setB8(<any>args + JSMarshalerArgumentOffsets.ReceiverShouldFree, true);
```
Comment (kg) [inline src/mono/browser/runtime/marshal.ts:109]: could be good to set all 4 bytes just for the sake of safety, but this is fine

File: src/mono/browser/runtime/memory.ts [hunk 3]
```diff
@@ -177,12 +186,17 @@ export function setF64(offset: MemOffset, value: number): void {
     Module.HEAPF64[<any>offset >>> 3] = value;
 }
 
-
+/** note: MonoBoolean is 8 bits not 32 bits when inside a structure or array */
 export function getB32(offset: MemOffset): boolean {
     receiveWorkerHeapViews();
     return !!(Module.HEAP32[<any>offset >>> 2]);
```
Comment (kg) [inline src/mono/browser/runtime/memory.ts:192]: as mentioned elsewhere i'd like to see this assert/warn if it encounters a value that isn't 0 or 1, but this is fine

**Review by kg — APPROVED** [review]

File: src/mono/browser/runtime/memory.ts [hunk 4]
```diff
@@ -177,10 +187,22 @@ export function setF64(offset: MemOffset, value: number): void {
...
 }
 
+let warnDirtyBool = true;
 
+/** note: MonoBoolean is 8 bits not 32 bits when inside a structure or array */
 export function getB32(offset: MemOffset): boolean {
     receiveWorkerHeapViews();
-    return !!(Module.HEAP32[<any>offset >>> 2]);
+    const value = (Module.HEAP32[<any>offset >>> 2]);
+    if (value > 1 && warnDirtyBool) {
```
Comment (kg) [inline src/mono/browser/runtime/memory.ts:196]: value could also be negative since it's heap32 and not heapu32

**Pull request merged**